workshops/models.py

```diff
@@ -227,9 +228,14 @@ def ongoing_events(self):
         return queryset
 
     def unpublished_events(self):
-        '''Return a queryset for events that are not yet published.'''
+        '''Return events without URLs that are upcoming or have unknown starts.
```

Comment:
And you'll also want to add something to this effect in the `help_text` for `Event.start` too, since you'll also need to have that set before the event is published.  A nice UI if we want to setup the right JavaScript would be to show a “published? {yes|no}” label, and have the “yes” case displayed dynamically when the edit form was configured in a way that would publish the site.  Then folks wouldn't have to track the underlying logic mentally.  But maybe that should wait until we have a better framework for JavaScript tooling.